bool appear
int16 point
int16 cx
int16 cy